client/src/modules/cohort/components/Cohorts.jsx

```diff
@@ -62,15 +80,27 @@ class Cohorts extends Component {
       fetchArchivedCohortsMetaData,
       removeArchivedCohortsMetaData
     } = this.props;
-    const action = areArchivedCohortsVisible
-      ? fetchArchivedCohortsMetaData
-      : removeArchivedCohortsMetaData;
-    action();
+
+    if (areArchivedCohortsVisible) {
+      this.setState({ pendingForArchivedCohorts: true }, () => {
+        fetchArchivedCohortsMetaData()
+          .then(() => this.setState({ pendingForArchivedCohorts: false }))
+          .catch(() => this.setState({ pendingForArchivedCohorts: false }));
+      });
+    } else {
```

Comment:
check if this else is necessary

Reply:
In my opinion, in this case, it's necessary. This method has to decide what to do depending on areArchivedCohortsVisible variable, but in one case we have to do more than one thing so I had to use if else statement.